### DISCUSSION TITLE
Documentation for newFrameStreamUnixLogger() framestream library options

### DISCUSSION THREAD

**Discussion by andrewimeson** [PowerDNS/pdns#14510] [Q&A] [answered]
The [documentation for `newFrameStreamUnixLogger()`](https://dnsdist.org/reference/dnstap.html#newFrameStreamUnixLogger) notes this:

> The following options apply to the settings of the framestream library. **Refer to the documentation of that library** for the default and allowed values for these options, as well as their exact descriptions.

I can't find the documentation, and I'm not sure which framestream library is being used. Could someone point me in the right direction? I'll make a PR to link to the docs if someone can provide them.

**Comment by rgacogne** [comment, marked as answer]
Hi Andrew,

You are completely right that the documentation should mention the exact library, and likely a link to the actual documentation it refers to. The library is https://github.com/farsightsec/fstrm and the documentation is available at http://farsightsec.github.io/fstrm/ Unfortunately it looks like the only place listing all options and their description is http://farsightsec.github.io/fstrm/group__fstrm__iothr.html#gaf594e508efcc7a5494af9b649db88d65 which is not very practical :-/

**Reply by andrewimeson** [reply]
You are right, those docs aren't particularly practical. It seems like the defaults are scattered about too, they're not actually shown in the function definition?

I'll at least send a PR linking to the library.

I have some ideas that aren't particularly great:
1. Manually extract and document the value of each configurable fstrm option that PowerDNS uses
2. Override the defaults in PowerDNS, and then document those overridden defaults

That's all I can think of.
